lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -533,6 +543,15 @@ public function addSelect($select = null/*, string ...$selects*/)
             return $this;
         }
 
+        if (is_array($select)) {
+            Deprecation::triggerIfCalledFromOutside(
```

Comment:
`QueryBuilder` isn't used by the DBAL, its API is only called from tests, should be `trigger()`.